R2API.Rules/RuleCatalogExtras.cs

```diff
@@ -0,0 +1,339 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Mono.Cecil.Cil;
+using MonoMod.Cil;
+using R2API.AutoVersionGen;
+using R2API.MiscHelpers;
+using RoR2;
+
+/*
+ * For Future Devs:
+ * Most public methods contain 2 local static methods, with the suffix DirectlyToCatalog & Internal
+ * DirectlyToCatalog is ran after the RuleCatalog gets initialized, and as the name implies, adds the new RuleDef or RuleCategory to the catalog instead of our collections.
+ * Internal is run before the RuleCatalog gets initialized, and as the name implies, it adds the new RuleDef or RuleCategory to our internal collections, which will be added later once the RuleCatalog gets initialized.
+ * 
+ * This is done for a number of reasons, the main one being that people can create new rules before the catalog gets initialized, so trying to add the rules to the nonexistent catalogs can cause issues, while some might want to add rules after the catalog initializes.
+ */
+namespace R2API;
+
+/// <summary>
+/// A class for adding new RuleDefs and RuleCategories to the game's RuleCatalog
+/// </summary>
+#pragma warning disable CS0436 // Type conflicts with imported type
+[AutoVersion]
+#pragma warning restore CS0436 // Type conflicts with imported type
+public static partial class RuleCatalogExtras
+{
+    /// <summary>
+    /// The Rules submodule's plugin GUID
+    /// </summary>
+    public const string PluginGUID = R2API.PluginGUID + ".rules";
+    /// <summary>
+    /// The Rules submodule's plugin name
+    /// </summary>
+    public const string PluginName = R2API.PluginName + ".Rules";
+
+    /// <summary>
+    /// An index that represents the Difficulty RuleCategoryDef Index
+    /// </summary>
+    public const int DifficultyCategoryIndex = 0;
```

Comment:
Use properties so that mods depending on them don't need to compile if they change